### PR TITLE
`General`: Show error messages for empty courses during export

### LINKED ISSUE #7288: `General`: Archiving an empty course loads forever
### Describe the bug

Archiving an empty course (no exercises) does load forever.

### To Reproduce

1. Create a new course
2. Don't add anything to it
3. Set the course start and end date to the past
4. Click on "Archive course"
5. It loads forever.

### Expected behavior

The course is archived.

### Screenshots

https://github.com/ls1intum/Artemis/assets/9006596/2b07de3d-bb24-4dad-ad6d-69123a8e16cf

### Which version of Artemis are you seeing the problem on?

6.5.3

### What browsers are you seeing the problem on?

Chrome

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by milljoniaer**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/server/).
- [x] I documented the Java code using JavaDoc style.
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
Closes #7288. Empty Courses seemed to be loading forever during archiving and showed a loading spinner, because the exercise task had been skipped and therefore the "complete" message was not sent.

### Description
This PR adds messages in the case of error, so that the client knows whenever the archiving fails and shows an error alert.

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Navigate to Course Administration
3. Create New Course with End Date before today.
4. Click the "Archive Course" Button and verify that there is an error displayed.

#### Exam Mode Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Exam with a Programming Exercise

1. Log in to Artemis
2. Participate in the exam as a student
3. Make sure that the UI of the programming exercise in the exam mode stays unchanged. You can use the [exam mode documentation](https://ls1intum.github.io/Artemis/user/exam_mode/) as reference.
5. ...

#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Test Coverage
| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [CourseExamExportService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5603/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.export/CourseExamExportService.html) | 91% | ✅ |
| [course-exam-archive-button.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5603/latest/artifact/shared/Coverage-Report-Client-Tests/app/shared/components/course-exam-archive-button/course-exam-archive-button.component.ts.html) | 87.93% | ✅|
### Screenshots
image

### CODE REVIEW

**Comment by aplr** [comment]
Hmm, could be weird if the UI is DE, as the server messages are always EN if I understand correctly? What about sending error codes, and doing i18n purely in the frontend? Or sending the message through the translation pipeline?

Also, I would show the course name, and not `Course {{id}}`.

**Comment by milljoniaer** [comment]
Thanks for the feedback, I implemented them, please have another look at the PR.

File: src/main/webapp/i18n/de/courseExamArchive.json [hunk 1]
```diff
@@ -6,8 +6,11 @@
             "archiveCourseSuccess": "Der Kurs wurde archiviert!",
             "archiveExamSuccess": "Die Prüfung wurde archiviert!",
             "archiveSuccessWithWarnings": "Der Archivierungsprozess wurde mit den folgenden Warnungen abgeschlossen:",
-            "archiveCourseError": "Der Kurs konnte nicht archiviert werden",
-            "archiveExamError": "Die Prüfung konnte nicht archiviert werden",
+            "archiveCourseError": {
+                "NOTHING_TO_EXPORT": "Der Kurs '{{ courseName }}' wurde nicht archiviert, da er leer ist.",
+                "DIR_NOT_CREATED": "Der Kurs '{{ courseName }}' konnte nicht archiviert werde: Der Ordner konnte nicht erstellt werden"
```
Comment (dearjasmina) [inline src/main/webapp/i18n/de/courseExamArchive.json:11]: Zeile 11: Hier fehlt "n": "archiviert werde**n**"

**Review by jakubriegel — COMMENTED** [review]
Please take a look a teamscale

**Review by jakubriegel — APPROVED** [review]

**Review by dearjasmina — APPROVED** [review]

**Review by valentin-boehm — APPROVED** [review]
lgtm

**Review by max-bergmann — APPROVED** [review]
Tested on TS3. LGTM

**Pull request merged**